### PR TITLE
Add `unsafeLazy` to simplify thread unsafe lazy initializations

### PR DESCRIPTION
**Pull request by Goooler**
Like: https://github.com/runningcode/kotlin-dsl/blob/master/subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/support/unsafeLazy.kt

### CODE REVIEW

File: base/src/main/java/app/tivi/extensions/Lazy.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app.tivi.extensions
+
+fun <T> unsafeLazy(initializer: () -> T): Lazy<T> = lazy(LazyThreadSafetyMode.NONE, initializer)
```
Comment (chrisbanes) [inline base/src/main/java/app/tivi/extensions/Lazy.kt:19]: We can make this an `inline` function too

**Review by chrisbanes — COMMENTED** [review]
Looks great overall

File: base/src/main/java/app/tivi/extensions/Lazy.kt (hunk 1 above)
Comment (Goooler) [reply base/src/main/java/app/tivi/extensions/Lazy.kt:19]: But also need to add `noinline` on `initializer`, should we do?

Comment (chrisbanes) [reply base/src/main/java/app/tivi/extensions/Lazy.kt:19]: Yeah sounds good. This function compiles down to what we had previously, just with a nicer API

**Review by chrisbanes — APPROVED** [review]
Thanks!

**Pull request merged**